java/com/google/flatbuffers/FlexBuffers.java

```diff
@@ -107,7 +109,7 @@
      * @return true if typed vector
      */
     static boolean isTypedVector(int type) {
-        return (type >= FBT_VECTOR_INT && type <= FBT_VECTOR_STRING) || type == FBT_VECTOR_BOOL;
+        return (type >= FBT_VECTOR_INT && type <= FBT_VECTOR_KEY) || type == FBT_VECTOR_BOOL;
```

Comment:
Shouldn't this still return true for existing data serialized this way?

Reply:
Internally it is okay for `FBT_VECTOR_STRING_DEPRECATED` not be a vector type, but 
`Reference::isTypedVector`, which is the only public API, would return false indeed.

Alright, I will change the logic.